.clang-format

```diff
@@ -0,0 +1,123 @@
+# -*- mode: yaml -*-
+---
+Language:        Cpp
+AccessModifierOffset: -4
+AlignAfterOpenBracket: AlwaysBreak
+AlignConsecutiveAssignments: false
+AlignConsecutiveDeclarations: false
+AlignEscapedNewlines: Right
+AlignOperands:   true
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: All
+AllowShortLambdasOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: WithoutElse
+AllowShortLoopsOnASingleLine: true
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: true
+AlwaysBreakTemplateDeclarations: true
+BinPackArguments: true
+BinPackParameters: false
+BraceWrapping:   
+  AfterCaseLabel:  false
+  AfterClass:      true
+  AfterControlStatement: false
+  AfterEnum:       false
+  AfterFunction:   true
+  AfterNamespace:  false
+  AfterObjCDeclaration: false
+  AfterStruct:     true
+  AfterUnion:      true
+  AfterExternBlock: false
+  BeforeCatch:     false
+  BeforeElse:      false
+  IndentBraces:    false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: true
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakInheritanceList: BeforeComma
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeComma
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     80
```

Comment:
Do we want this strict of a line length restriction?  Strict enforcement of an 80 column limit can lead to harder to read formatting.  I personally prefer setting this value to 0 and enforcing line limits during code review based on situational style preferences and readability.  If we do want to enforce a hard cap, I'd be in support of setting the column limit to a higher value (say 200) to allow some flexibility.

Reply:
This was taken directly from [AMR-Wind](https://github.com/Exawind/amr-wind/blob/main/.clang-format). I agree that 80 characters is pretty restrictive, but I'll defer these decisions to those who will be working in this code directly in the future. There are other decisions that impact this such as variable and namespace conventions. For example, I called the primary namespace `openturbine::` which itself is pretty long and takes up a large portion of an 80 column limit.

Reply:
This is the same file as used in AMR-Wind, Nalu-Wind. and the exawind-driver. This is the only objection we've had to it so far. I don't really have a preference, but if this is an issue, we can address it later in all projects.

Reply:
I know that we've had the 80 character limit in Nalu-Wind lead to some less ideal line breaks when working on it.  This is especially true when working with longer variable names and some of C++'s more verbose syntax.  In talking with some other STK guys, 120 or 150 seems like a more practical limit.  There's no need to come up with a definitive answer now - just something to keep in mind.